Backend/Remora.Discord.Rest/API/Channels/DiscordRestChannelAPI.cs

```diff
@@ -1179,4 +1181,96 @@ public virtual Task<Result<IReadOnlyList<IThreadMember>>> ListThreadMembersAsync
             ct: ct
         );
     }
+
+    /// <inheritdoc />
+    public virtual Task<Result<IChannelThreadQueryResponse>> ListPublicArchivedThreadsAsync
+    (
+        Snowflake channelID,
+        Optional<DateTimeOffset> before = default,
+        Optional<int> limit = default,
+        CancellationToken ct = default
+    )
+    {
+        return this.RestHttpClient.GetAsync<IChannelThreadQueryResponse>
+        (
+            $"channels/{channelID}/threads/archived/public",
+            b =>
+            {
+                if (before.HasValue)
+                {
+                    TimeSpan offset = before.Value.Offset;
+                    b.AddQueryParameter("before", before.Value.ToString($"yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'ffffff'+'{offset.Hours:D2}':'{offset.Minutes:D2}"));
```

Comment:
It might be more appropriate to break this out into a helper such as an extension method, since it's used in multiple places.